frappe/desk/reportview.py

```diff
@@ -240,16 +240,19 @@ def delete_items():
 		delete_bulk(doctype, items)
 
 def delete_bulk(doctype, items):
-	failed = []
 	for i, d in enumerate(items):
 		try:
 			frappe.delete_doc(doctype, d)
 			if len(items) >= 5:
 				frappe.publish_realtime("progress",
 					dict(progress=[i+1, len(items)], title=_('Deleting {0}').format(doctype), description=d),
 						user=frappe.session.user)
+			# Commit after successful deletion
+			frappe.db.commit()
 		except Exception:
-			failed.append(d)
+			# rollback if any record failed to delete
+			# if not rollbacked, queries get committed on after_request method in app.py
+			frappe.db.rollback()
```

Comment:
I guess this `rollback` will not affect deletion of dynamic links which was enqueued in `delete_doc`.

In case of error, `delete_dynamic_links` will still get executed right?

Reply:
That's why I have moved `delete_dynamic_links` at the end of block after all validations. So, if there are any errors, it does not get executed.